apps/client/src/features/app-settings/panel/project-panel/ProjectListItem.tsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from 'react';
+import { IconButton, Menu, MenuButton, MenuItem, MenuList } from '@chakra-ui/react';
+import { IoEllipsisHorizontal } from '@react-icons/all-files/io5/IoEllipsisHorizontal';
+
+import { invalidateAllCaches, maybeAxiosError } from '../../../../common/api/apiUtils';
+import { deleteProject, duplicateProject, loadProject, renameProject } from '../../../../common/api/ontimeApi';
+
+import ProjectForm, { ProjectFormValues } from './ProjectForm';
+import { EditMode } from './ProjectList';
+
+import style from './ProjectPanel.module.scss';
+
+interface ProjectListItemProps {
+  current?: boolean;
+  filename: string;
+  createdAt: string;
+  updatedAt: string;
+  onToggleEditMode: (editMode: EditMode, filename: string | null) => void;
+  onSubmit: () => void;
+  onRefetch: () => Promise<void>;
+  editingFilename: string | null;
+  editingMode: EditMode | null;
+}
+
+export default function ProjectListItem({
+  current,
+  createdAt,
+  editingFilename,
+  editingMode,
+  filename,
+  onRefetch,
+  onSubmit,
+  onToggleEditMode,
+  updatedAt,
+}: ProjectListItemProps) {
+  const [submitError, setSubmitError] = useState<string | null>(null);
+
+  const handleSubmitRename = async (values: ProjectFormValues) => {
+    try {
+      setSubmitError(null);
+
+      if (!values.filename) {
+        setSubmitError('Filename cannot be blank');
+        return;
+      }
+      await renameProject(filename, values.filename);
+      await onRefetch();
+      onSubmit();
+    } catch (error) {
+      setSubmitError(maybeAxiosError(error));
+    }
+  };
+
+  const handleSubmitDuplicate = async (values: ProjectFormValues) => {
+    try {
+      setSubmitError(null);
+
+      if (!values.filename) {
+        setSubmitError('Filename cannot be blank');
+        return;
+      }
+      await duplicateProject(filename, values.filename);
+      await onRefetch();
```

Comment:
question: Don't we `refetch` as well in `<ProjectList>` in the `handleSubmit` function? Who should be responsible for fetching? maybe `<ProjectList>` since it is the one that provides the data?

Reply:
`ProjectListItem` handles submission for `duplicate` and `rename`, and `ProjectPanel` for `create`.

`ProjectList` is currently responsible for fetching

Reply:
hm, so why are we calling a refetch here?

Reply:
not sure I understand, as of now with the changes:

`ProjectList` deals only with `create`, and calls a refetch for it
`ProjectListItem` handles `duplicate` and `rename`, so it calls the `refetch` callback after we made the query so it refreshes the list

Reply:
What I mean is that the fetching of the list is not in the scope of the `ProjectListItem`.

We already have the `onSubmit` callback which allows the parents to handle side effects, why not make the parent (which is responsible for fetching) call the refetch?